src/config_manager.ts

```diff
@@ -24,17 +39,40 @@ export class ConfigManager {
      */
     public getTypedRegexes() {
         // Access the configuration
-        let regexArray : any = vscode.workspace.getConfiguration('live-notebook').get('regexes');
-
-        // Create a new array of typed regular expressions,
-        // and assign them to the configuration values
-        let regexes = new Array<TypedRegex>(regexArray.length);
-        for (let i = 0; i < regexArray.length; i++) {
-            regexes[i] = {type: regexArray[i].type, regex: regexArray[i].regex};
-        }
+        let regexArray: any = this.configuration.get('regexes');
         return regexArray;
     }
 
-    // TODO API config accessing code
+    public onDidUpdateConfiguration(onUpdate: (value: any) => unknown | undefined | void) {
+        this.onUpdates.push(onUpdate);
+    }
+
+    /**
+     * Access the config file in package.json
+     * and load in the APIs as their respective calling
+     * strategies/methods.
+     * @returns An array of API strategies
+     */
+    public getAPIStrategies() {
+        // Access the configuration
+        let apiArray: any = this.configuration.get('apis');
+
+        // Create a new array of api strategies
+        let strategies = new Array<APIStrategy>(apiArray.length);
+        for (let i = 0; i < strategies.length; i++) {
+            switch (apiArray[i].method) {
+                case "POST":
+                    strategies[i] = new PostStrategy(apiArray[i]);
+                    break;
+                case "GET":
+                    strategies[i] = new GetStrategy(apiArray[i]);
+                    break;
+                case "VirusTotal":
```

Comment:
Did you ever determine if other relevant APIs behavior similarly to VirusTotal? If so, this could be more generic.

Reply:
I talked about it is possible to chain POST/GET calls to emulate VirusTotal a few weeks ago but @PrateekVachher brought up that each call to VirusTotal is queued such that one might get a successful response that does not contain the full requested information, so a more generalize chaining strategy, though possible, still would not be very useful for VT.

Reply:
Like Zed said, virustotal seems to be pretty unique, at least from the apis I have seen. It requires us to store state client side (the key for the actual scan results), which, if I understand correctly, means it is not a RESTful API, while all of the other APIs I have seen are.